### ISSUE TITLE
[BUG] Pydantic Flytekit plugin is not compatible with Pydantic v2

### ISSUE THREAD

**Issue by adartaud** [flyteorg/flyte#4603]
### Describe the bug

Since Pydantic v2, the method `__get_validators__` has been removed from `pydantic.BaseModel`. However as you can see [here](https://github.com/flyteorg/flytekit/blob/master/plugins/flytekit-pydantic/flytekitplugins/pydantic/deserialization.py#L12) `flytekitplugins-pydantic` still uses it.
This raises `AttributeError: __get_validators__`

### Expected behavior

At the moment, there are no constraints on the version of Pydantic in the [setup.py](https://github.com/flyteorg/flytekit/blob/master/plugins/flytekit-pydantic/setup.py) of the plugin, although it only seems to be compatible with versions prior to v2.
`flytekitplugins-pydantic` should be compatible with new versions of Pydantic, or at least specify a constraint while this isn't fixed.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by eapolinario** [comment]
@adartaud , thanks for the report! 

We raised a [concern](https://github.com/flyteorg/flytekit/pull/1660/files#r1258878629) about pydantic v2 in https://github.com/flyteorg/flytekit/pull/1660, but it looks like at the time pydantic v2 could be supported. We should validate and increase test coverage if possible. 

@adartaud , do you have a repro?

**Comment by adartaud** [comment]
Thanks for your reply.
I tested again with up-to-date versions and I still encounter the issue. 

Here is a repro:

1. Create and activate a new virtual env (`python -m venv env` &  `source env/bin/activate`)
2. Install these dependencies with `pip install -r requirements.txt`

```txt
# requirements.txt
flytekit==1.10.2
flytekitplugins-pydantic==1.10.2
pydantic==2.5.2
```

3. Run a python shell and simply try to run `import flytekit`. This will raise `AttributeError: __get_validators__`

**Comment by davidmirror-ops** [comment]
@adartaud I can repro the behavior you describe here. Nevertheless, when I use the recent `flytekitplugins-pydantic-1.11.0` then it works, following the example described in [2217](https://github.com/flyteorg/flytekit/pull/2217)

Can you verify and confirm if it works for you?

**Comment by eapolinario** [comment]
@Future-Outlier , can you confirm that this is no longer the case and close this issue?

**Comment by Future-Outlier** [comment]
> @Future-Outlier , can you confirm that this is no longer the case and close this issue?

yes we can close this, and also pydantic v1 is not supported now, we only support pydantic v2.
